cmd/snap-confine/mount-support.c

```diff
@@ -743,3 +743,45 @@ void sc_setup_user_mounts(struct sc_apparmor *apparmor, int snap_update_ns_fd,
 	sc_do_mount("none", "/", NULL, MS_REC | MS_SLAVE, NULL);
 	sc_call_snap_update_ns_as_user(snap_update_ns_fd, snap_name, apparmor);
 }
+
+void sc_ensure_snap_dir_shared_mounts(void)
+{
+	const char *dirs[] = { SNAP_MOUNT_DIR, "/var/snap", NULL };
+	for (int i = 0; dirs[i] != NULL; i++) {
+		const char *dir = dirs[i];
+		if (!is_mounted_with_shared_option(dir)) {
+			/* Since this directory isn't yet shared (but it should be),
+			 * recursively bind mount it, then recursively share it so that
+			 * changes to the host are seen in the snap and vice-versa. This
+			 * allows us to fine-tune propagation events elsewhere for this new
+			 * mountpoint. */
+			sc_do_mount(dir, dir, "none", MS_BIND | MS_REC, 0);
```

Comment:
Can you add one more bit of information here: The mount options configured here differ from those that are set up by the snap.mount unit, on purpose.